File: packages/cli/templates/project/function-websocket.py
import json
from typing import Any


def {{HANDLER}}(event: dict[str, Any], context: Any) -> dict[str, Any]:
    request_context = event.get("requestContext", {})
    event_type = request_context.get("eventType")
    if event_type != "MESSAGE":
        return {"statusCode": 200, "body": ""}

    return {
        "statusCode": 200,
        "headers": {"content-type": "application/json; charset=utf-8"},
        "body": json.dumps(
            {
                "ok": True,
                "connectionId": request_context.get("connectionId"),
                "messageId": request_context.get("messageId"),
                "requestId": getattr(context, "request_id", None),
                "body": event.get("body", ""),
            }
        ),
    }

File: packages/cli/templates/project/function-websocket.go
package main

import (
	"context"
	"encoding/json"
)

type WebSocketRequestContext struct {
	ConnectionID string `json:"connectionId"`
	EventType    string `json:"eventType"`
	MessageID    string `json:"messageId,omitempty"`
}

type WebSocketEvent struct {
	Body           string                  `json:"body"`
	RequestContext WebSocketRequestContext `json:"requestContext"`
}

type WebSocketResponse struct {
	StatusCode int               `json:"statusCode"`
	Headers    map[string]string `json:"headers,omitempty"`
	Body       string            `json:"body"`
}

func {{HANDLER}}(ctx context.Context, event WebSocketEvent) (WebSocketResponse, error) {
	if event.RequestContext.EventType != "MESSAGE" {
		return WebSocketResponse{StatusCode: 200, Body: ""}, nil
	}

	requestID, _ := ctx.Value("lambdaRuntimeRequestID").(string)
	body, err := json.Marshal(map[string]any{
		"ok":           true,
		"connectionId": event.RequestContext.ConnectionID,
		"messageId":    event.RequestContext.MessageID,
		"requestId":    requestID,
		"body":         event.Body,
	})
	if err != nil {
		return WebSocketResponse{}, err
	}
	return WebSocketResponse{
		StatusCode: 200,
		Headers:    map[string]string{"content-type": "application/json; charset=utf-8"},
		Body:       string(body),
	}, nil
}
